camel/model_backend.py

```diff
@@ -0,0 +1,101 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+from camel.typing import ModelType
+
+
+class ModelBackend(ABC):
+    r"""Base class for different model backends.
+    May be OpenAI API, a local LLM, a stub for unit tests, etc."""
+
+    @abstractmethod
+    def run(self, *args, **kwargs) -> Dict[str, Any]:
+        r"""Runs the query to the backend model.
+
+        Raises:
+            RuntimeError: if the return value from OpenAI API
+            is not a dict that is expected.
+
+        Returns:
+            Dict[str, Any]: All backends must return a dict in OpenAI format.
+        """
+        pass
+
+
+class OpenAIModel(ModelBackend):
+    r"""OpenAI API in a unified ModelBackend interface."""
+
+    def __init__(self, model_type: ModelType, model_config_dict: Dict) -> None:
+        super().__init__()
+        self.model_type = model_type
+        self.model_config_dict = model_config_dict
+
+    def run(self, *args, **kwargs) -> Dict[str, Any]:
```

Comment:
Should we add type hinting for args and kwargs?

Reply:
Yes, will do in the next PR